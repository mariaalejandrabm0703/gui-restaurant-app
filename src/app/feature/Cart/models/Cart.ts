export interface IClient {
    /**
     * Identificador del cliente
     */
    id?: number;
    /**
     * Nombre del cliente
     */
    nombre: string;
    /**
     * Identificacion del cliente
     */
    identificacion: string;
    /**
     * Telefono del cliente
     */
    telefono: string;
    /**
     * Estado del producto
     */
    activo?: string;
    /**
     * Email del cliente
     */
    email: string;
  }