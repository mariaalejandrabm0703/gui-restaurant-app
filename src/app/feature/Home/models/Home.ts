export interface IProduct {
  /**
   * Identificador del producto
   */
  id: number;
  /**
   * Descripción del producto
   */
  descripcion: string;
  /**
   * Categoría del producto
   */
  categoria: string;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Estado del producto
   */
  activo: string;
  /**
   * Imagen del producto
   */
  img: string;
}

export interface IProductOrderRegistred {
  /**
   * Identificador del producto
   */
  id: number;
 /**
   * Cantidad del producto
   */
  cantidad: number;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Estado del producto
   */
  activo: string;
  /**
   * Producto del pedidoProducto
   */
  producto: IProduct;
}

export interface IProductOrder {
  /**
   * Identificador del producto
   */
   producto: number;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Cantidad del producto
   */
  cantidad: number;
}
export interface IRanking {
  /**
   * Identificador del producto
   */
  id: number;
  /**
   * Descripción del producto
   */
  descripcion: string;
  /**
   * Categoría del producto
   */
  categoria: string;
  /**
   * Precio del producto
   */
  precio: number;
  /**
   * Estado del producto
   */
  activo: string;
  /**
   * Imagen del producto
   */
  img: string;
  /**
   * Cantidad vendida del producto
   */
  cantidad: number;
}
