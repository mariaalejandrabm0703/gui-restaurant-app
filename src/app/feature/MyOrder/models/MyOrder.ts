import { IProductOrder } from 'app/feature/Home/models/Home';

export interface IMyClient {
  /**
   * Identificador del cliente
   */
  id: number;
  /**
   * Nombre del cliente
   */
  nombre: string;
  /**
   * Indetificación del cliente
   */
  identificacion: string;
  /**
   * Telefono del cliente
   */
  telefono: string;
  /**
   * Email del cliente
   */
  email: string;
  /**
   * Estado del cliente
   */
  activo: string;
}
export interface IMyOrder {
  /**
   * Identificador del pedido
   */
  id: number | 0 ;
  /**
   * Descripción del pedido
   */
  fechaEntrega: string;
  /**
   * Precio del pedido
   */
  precio: number;
  /**
   * Estado del pedido
   */
  activo: string;
  /**
   * Productos
   */
  productosPedidos: IProductOrder[];
  /**
   * cliente del pedido
   */
  cliente: IMyClient;
}

export interface IMyOrderReg {
  /**
   * Descripción del pedido
   */
  fechaEntrega: string;
  /**
   * Precio del pedido
   */
  precio: number;
  /**
   * Estado del pedido
   */
  activo: string;
  /**
   * Productos
   */
  productos: IProductOrder[];
  /**
   * cliente del pedido
   */
  cliente: number;
}