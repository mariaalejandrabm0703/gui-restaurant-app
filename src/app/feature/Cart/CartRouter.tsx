import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from '../../shared/components/LazyFallback';

const CartPage = React.lazy(() => import('./pages/Cart'));

export const CartRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    <Switch>
      <Route path="/" component={CartPage}></Route>
    </Switch>
  </React.Suspense>
);
