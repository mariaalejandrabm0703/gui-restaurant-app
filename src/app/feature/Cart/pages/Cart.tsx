import * as React from 'react';
import { Layout } from 'app/shared/components/Layout';
import { ManageCart } from '../hoc/ManageCart';
import { RouteComponentProps } from 'react-router-dom';

const CartPage: React.FC<RouteComponentProps> = () => (
  <Layout title="Carrito" description="Carrito del cliente.">
    <ManageCart>Hola9</ManageCart>
  </Layout>
);

CartPage.displayName = 'CartPage';

export default CartPage;
