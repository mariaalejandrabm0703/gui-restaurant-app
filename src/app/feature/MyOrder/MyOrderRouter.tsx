import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from '../../shared/components/LazyFallback';

const MyOrderPage = React.lazy(() => import('./pages/MyOrderPage'));

export const MyOrderRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    <Switch>
      <Route path="/myOrder" component={MyOrderPage}></Route>
    </Switch>
  </React.Suspense>
);
