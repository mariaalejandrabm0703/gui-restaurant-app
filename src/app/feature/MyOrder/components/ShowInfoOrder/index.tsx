import * as PropTypes from 'prop-types';
import { IMyOrder, IMyOrderReg } from '../../models/MyOrder';
import React, { useEffect, useState } from 'react';
import {
  setClient,
  setCountProduct,
  setProducts,
} from '../../../../core/redux/actions/cart/ActionsCart';
import {
  setConfigOrderAsync,
  setOrder,
} from '../../../../core/redux/actions/order/ActionsOrder';
import { IClient } from 'app/feature/Cart/models/Cart';
import { IProduct } from 'app/feature/Home/models/Home';
import { IProductOrder } from '../../../Home/models/Home';
import { Istate } from 'app/core/redux/modelo/GeneralState';
import { ManageCardProduct } from 'app/shared/components/cardProduct/cardProduct';
import { connect } from 'react-redux';
import { currencyFormat } from 'app/shared/utils/utils';
import { useHistory } from 'react-router';

interface ShowInfoOrderProps {
  myOrder: IMyOrder;
  listProducts: Array<IProduct>;
  setConfigOrderAsync: (orderSet: IMyOrder, orderEdit: IMyOrderReg) => void;
  setClient: (client: IClient) => void;
  setCountProduct: (count: number) => void;
  setProducts: (products: Array<IProductOrder>) => void;
  setOrder: (pedido: IMyOrder) => void;
}

const ShowInfoOrder: React.FC<ShowInfoOrderProps> = ({
  myOrder,
  listProducts,
  setConfigOrderAsync,
  setClient,
  setCountProduct,
  setProducts,
  setOrder,
}) => {
  const [productos, setproductos] = useState([
    { id: 0, desc: '', price: 0, img: '', cantidad: 0 },
  ]);

  const history = useHistory();

  useEffect(() => {
    const newProducts = [];
    for (const p of myOrder.pedidosProductos) {
      const newProds = listProducts.filter((item) => item.id === p.producto.id);
      if (newProds.length > 0) {
        newProducts.push({
          id: p.id,
          desc: newProds[0].descripcion,
          price: p.precio,
          img: newProds[0].img,
          cantidad: p.cantidad,
        });
      }
    }
    setproductos(newProducts);
  }, [myOrder]);

  const handleSet = () => {
    const productsOrderCart = Array<IProductOrder>();
    let cantidad: number = 0;
    for (const item of myOrder.pedidosProductos) {
      cantidad = cantidad + item.cantidad;
    }
    setCountProduct(cantidad);

    for (const item of myOrder.pedidosProductos) {
      productsOrderCart.push({
        producto: item.producto.id,
        precio: item.precio,
        cantidad: item.cantidad,
      });
    }
    setProducts(productsOrderCart);

    const client: IClient = myOrder.cliente;
    setClient(client);
    // borrar store myOrder
    setOrder({
      id: 0,
      fechaEntrega: '',
      precio: 0,
      activo: '',
      pedidosProductos: [],
      cliente: {
        id: 0,
        nombre: '',
        identificacion: '',
        telefono: '',
        email: '',
        activo: '',
      },
    });
    history.push('/cart');
  };

  const handleCancel = () => {
    // activo = 0 cero
    const myOrderSet = { ...myOrder, activo: '0' };

    const productosOrder = myOrder.pedidosProductos.map((item) => {
      return {
        producto: item.producto.id,
        precio: item.cantidad * item.producto.precio,
        cantidad: item.cantidad,
      };
    });
    const myOrderReg: IMyOrderReg = {
      fechaEntrega: myOrder.fechaEntrega,
      precio: myOrder.precio,
      activo: '0',
      cliente: myOrder.cliente.id,
      productos: productosOrder,
    };
    setConfigOrderAsync(myOrderSet, myOrderReg);
  };

  return (
    <div>
      <div className="container d-flex">
        <div className="align-items-end m-2">
          <h5>Informaci√≥n de pedido #{myOrder.id}:</h5>
        </div>
        <div className="align-items-center">
          <a type="submit" onClick={handleCancel} className="btn btn-link">
            Cancelar
          </a>
        </div>
        <div className="align-items-center">
          <a type="submit" onClick={handleSet} className="btn btn-link">
            Editar
          </a>
        </div>
      </div>
      <div>
        <p>Estado: {myOrder.activo === '1' ? 'Activo' : 'Cancelado'}</p>
        <p>Precio: {currencyFormat(myOrder.precio)} </p>
        <p>Fecha de entrega: {myOrder.fechaEntrega} </p>
      </div>
      <div>
        <p>Cliente: #{myOrder.cliente.identificacion}</p>
        <p>Nombre: {myOrder.cliente.nombre}</p>
      </div>
      <div>
        <p>Productos:</p>
        <div className="container d-flex row-cards">
          {productos.map((product) =>
            product.desc != '' ? (
              <div key={product.id} className="m-3">
                <ManageCardProduct
                  key={product.id}
                  id={product.id}
                  description={product.desc}
                  price={product.price}
                  img={product.img}
                  showAdd={false}
                />
              </div>
            ) : null
          )}
        </div>
      </div>
    </div>
  );
};
ShowInfoOrder.propTypes = {
  listProducts: PropTypes.array.isRequired,
  myOrder: PropTypes.shape({
    id: PropTypes.number.isRequired,
    fechaEntrega: PropTypes.string.isRequired,
    precio: PropTypes.number.isRequired,
    activo: PropTypes.string.isRequired,
    pedidosProductos: PropTypes.array.isRequired,
    cliente: PropTypes.shape({
      id: PropTypes.number.isRequired,
      nombre: PropTypes.string.isRequired,
      identificacion: PropTypes.string.isRequired,
      telefono: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired,
      activo: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  setConfigOrderAsync: PropTypes.func.isRequired,
  setClient: PropTypes.func.isRequired,
  setCountProduct: PropTypes.func.isRequired,
  setProducts: PropTypes.func.isRequired,
  setOrder: PropTypes.func.isRequired,
};

const mapStateToProps = (state: Istate) => {
  return {
    listProducts: state.products.listProducts,
  };
};

export default connect(mapStateToProps, {
  setConfigOrderAsync,
  setClient,
  setCountProduct,
  setProducts,
  setOrder,
})(ShowInfoOrder);
