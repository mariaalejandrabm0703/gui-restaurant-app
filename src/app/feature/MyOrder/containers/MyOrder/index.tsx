import * as PropTypes from 'prop-types';
import * as React from 'react';
import { IErrorToast } from '../../../../core/redux/modelo/IStateMain';
import { IMyOrder } from '../../models/MyOrder';
import ToastError from '../../../../shared/components/ToastError';
import {setOrderAsync} from '../../../../core/redux/actions/order/ActionsOrder';

interface OrderProps {
  myOrder: IMyOrder;
  isLoading: boolean;
  errorMessage: IErrorToast;
}

export const MyOrder: React.FC<OrderProps> = ({
  myOrder,
  isLoading,
   errorMessage,

}) => {
  // React.useEffect(() => {
  //   if (myOrder && errorMessage.message === '') {
  //     setOrderAsync(1);
  //   }
  // }, [myOrder]);

  return (
    <div className="container">
      <ToastError />
      <h1>Mi pedido</h1>
    </div>
  );
};
