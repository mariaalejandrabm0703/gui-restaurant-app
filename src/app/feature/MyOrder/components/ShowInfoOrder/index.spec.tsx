import '@testing-library/jest-dom';
import {
  IFilters,
  IProduct,
  IProductOrder,
  IProductOrderRegistred,
  IRanking,
} from '../../../Home/models/Home';
import { IMyClient, IMyOrder } from '../../models/MyOrder';
import {
  clientInfo,
  filtersInfo,
  myClientInfo,
  myOrderInfo,
  productInfo,
  productOrderInfor,
  productOrderRegistredInfo,
  rankingInfo,
} from '../../../../shared/utils/data';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { IClient } from '../../../Cart/models/Cart';
import { IStateCart } from '../../../../core/redux/modelo/IStateCart';
import { IStateHome } from '../../../../core/redux/modelo/IStateHome';
import { IStateMain } from '../../../../core/redux/modelo/IStateMain';
import { IStateOrder } from '../../../../core/redux/modelo/IStateOrder';
import { IStateProducts } from '../../../../core/redux/modelo/IStateProducts';
import { IStateRanking } from '../../../../core/redux/modelo/IStateRanking';
import { Provider } from 'react-redux';
import React from 'react';
import ReactDOM from 'react-dom';
import ShowInfoOrder from './index';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';

const mockStore = configureStore([]);

const newProduct: IProduct = productInfo;
const listProducts: IProduct[] = [newProduct];
const products: IStateProducts = {
  listProducts,
};

const newRamking: IRanking = rankingInfo;
const listRanking: IRanking[] = [newRamking];
const ranking: IStateRanking = {
  listRanking,
};

const count: number = 1;
const client: IClient = clientInfo;
const modo: string = 'registry';
const newProductOrder: IProductOrder = productOrderInfor;
const listProductsCart: IProductOrder[] = [newProductOrder];
const cart: IStateCart = {
  listProductsCart,
  count,
  client,
  modo,
};

const main: IStateMain = {
  isLoading: false,
  errorMessage: { message: '', type: '' },
};

const newProductRegistred: IProductOrderRegistred = productOrderRegistredInfo;
const pedidosProductos: IProductOrderRegistred[] = [newProductRegistred];
const cliente: IMyClient = myClientInfo;
const myOrder: IMyOrder = {
  id: 1,
  fechaEntrega: '01/10/2021',
  precio: 500,
  activo: '1',
  pedidosProductos,
  cliente: cliente,
};
const order: IStateOrder = {
  myOrder,
};

const listProductsHome: IProduct[] = [newProduct];
const filters: IFilters = filtersInfo;
const home: IStateHome = {
  listProductsHome,
  filters,
};

describe('Pruebas de componente ShowInfoOrder', () => {
  let store;

  let wrapper;
  beforeEach(() => {
    store = mockStore({
      products,
      ranking,
      cart,
      main,
      order,
      home,
    });

    wrapper = shallow(
      <Provider store={store}>
        <BrowserRouter>
          <ShowInfoOrder 
            myOrder={myOrder}
            />
        </BrowserRouter>
        ,
      </Provider>
    );
  });

  it('rederizar el componente ShowInfoOrder sin errores', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <BrowserRouter>
          <ShowInfoOrder myOrder={myOrder}/>
        </BrowserRouter>
        ,
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Compara snapshot del ShowInfoOrder renderizado', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
