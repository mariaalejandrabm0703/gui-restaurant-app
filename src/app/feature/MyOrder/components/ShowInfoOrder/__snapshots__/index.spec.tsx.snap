// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente ShowInfoOrder Compara snapshot del ShowInfoOrder renderizado 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <BrowserRouter>
    <Connect(ShowInfoOrder)
      myOrder={
        Object {
          "activo": "1",
          "cliente": Object {
            "activo": "1",
            "email": "maria@maria.com",
            "id": 1,
            "identificacion": "109049544",
            "nombre": "Maria",
            "telefono": "3042569874",
          },
          "fechaEntrega": "01/10/2021",
          "id": 1,
          "pedidosProductos": Array [
            Object {
              "activo": "1",
              "cantidad": 1,
              "id": 1,
              "precio": 500,
              "producto": Object {
                "activo": "1",
                "categoria": "Plato principal",
                "descripcion": "Pastas",
                "id": 1,
                "img": "https://cdns.iconmonstr.com/wp-content/assets/preview/2019/240/iconmonstr-product-3.png",
                "precio": 500,
              },
            },
          ],
          "precio": 500,
        }
      }
    />
  </BrowserRouter>
  ,
</ContextProvider>
`;
